scribus/text/textshaper.cpp

```diff
@@ -396,8 +395,9 @@ QList<GlyphCluster> TextShaper::shape()
 			if (textRun.dir == UBIDI_RTL)
 				run.setFlag(ScLayout_RightToLeft);
 
-			lineBoundery.setPosition(firstCluster);
-			if (lineBoundery.isAtBoundary())
+			BreakIterator* lineBoundery = getLineIterator();
+			assert(lineBoundery != NULL);
```

Comment:
Asserts generally are reserved for things that never expected to fail and we have code that assume they never fail and will crash anyway, so we crash early to make the source of the failure clear.